src/Microsoft.AspNet.Mvc.Core/ResponseEncodingWrapper.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Text;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Encoding wrapper which makes preamble to be not required.
+    /// </summary>
+    internal class ResponseEncodingWrapper : Encoding
```

Comment:
If we remove the usage from `ContentResult`, then make this a private type in `ViewExecutor`.